### ISSUE TITLE
ng-xi18n not working with sass

### ISSUE THREAD

**Issue by TsarS** [mgechev/angular-seed#1488]
I use scss
When i try to `npm run i18n` i getiing an error

```
  `Error: Compilation failed. Resource file not found: d:/OpenServer/domains/angular-      seed/src/client/app/shared/navbar/navbar.component.css
      at Object.get (d:\OpenServer\domains\angular-seed\node_modules\@angular\compiler-   cli\src\extract_i18n.js:126:27)
      at DirectiveNormalizer._fetch (d:\OpenServer\domains\angular-   seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13329:45)
      at d:\OpenServer\domains\angular-   seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13429:55
     at Array.map (native)
     at DirectiveNormalizer._loadMissingExternalStylesheets (d:\OpenServer\domains\angular-seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13429:16)
     at DirectiveNormalizer.normalizeExternalStylesheets (d:\OpenServer\domains\angular-seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13411:23)
     at d:\OpenServer\domains\angular-seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13360:70
     at process._tickCallback (internal/process/next_tick.js:103:7)
     at Module.runMain (module.js:606:11)
     at run (bootstrap_node.js:394:7)
   Extraction failed`
```

**Comment by mgechev** [comment]
@ruffiem would you take a look at this one?

**Comment by ruffiem** [comment]
@mgechev sure, I will try to reproduce the error tomorrow and come back with some update

**Comment by TsarS** [comment]
Temporary i figured it out by comment all `styleUrls`  `//  styleUrls: ['about.component.css']`  ))

**Comment by ruffiem** [comment]
Can't reproduce but I'am not sure this is related to the seed.

**Comment by mgechev** [comment]
@ruffiem thanks for trying this out!

**Comment by vlad0** [comment]
Hi I have the same issue on Windows machine. 
If I don't use SASS everything works fine with i18n but the moment I switch to SASS the `ng-xi18n` is looking for the referenced `styleUrls` in the components, e.g. `styleUrls: ['about.component.css']` 
Is there any pre-i18n command needed, something that will compile SASS files? 

Steps I did from scratch:
1. clone the angular2-seed repo (master)
2. enable SASS following the WIKI https://github.com/mgechev/angular-seed/wiki/Working-with-SASS
2.1. set ENABLE_SCSS = true; at seed.config.ts // line 289
3. enable i18n following the https://github.com/mgechev/angular-seed/wiki/Working-with-SASS

and get this error: 
`$ npm run i18n

> angular-seed@0.0.0 i18n D:\Angular2\angular2-seed
> ng-xi18n && gulp clean.i18n

Error: Compilation failed. Resource file not found: D:/Angular2/angular2-seed/src/client/app/shared/toolbar/toolbar.component.css
    at Object.resourceLoader.get (D:\Angular2\angular2-seed\node_modules\@angular\compiler-cli\src\extract_i18n.js:13:23)
    at DirectiveNormalizer._fetch (D:\Angular2\angular2-seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13661:45)
    at D:\Angular2\angular2-seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13761:55
    at Array.map (native)
    at DirectiveNormalizer._loadMissingExternalStylesheets (D:\Angular2\angular2-seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13761:16)
    at DirectiveNormalizer.normalizeExternalStylesheets (D:\Angular2\angular2-seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13743:23)
    at D:\Angular2\angular2-seed\node_modules\@angular\compiler\bundles\compiler.umd.js:13692:70
    at process._tickCallback (node.js:369:9)
    at Function.Module.runMain (module.js:443:11)
    at startup (node.js:139:18)
Extraction failed

npm ERR! Windows_NT 10.0.10586
npm ERR! node v4.4.5
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! angular-seed@0.0.0 i18n: `ng-xi18n && gulp clean.i18n`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the angular-seed@0.0.0 i18n script 'ng-xi18n && gulp clean.i18n'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the angular-seed package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ng-xi18n && gulp clean.i18n
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs angular-seed
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls angular-seed
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\Angular2\angular2-seed\npm-debug.log
`

**Comment by ruffiem** [comment]
I am using a custom SASS implementation on my seed and it works well. I have never tried the seed SASS support. It a problem between the seed SASS support and the ng-xi18n.

@mgechev want me to take a look at this ?

**Comment by mgechev** [comment]
@ruffiem if you have some time it'll be great.

**Comment by sebastianhaas** [comment]
I can confirm this, `ng-xi18n` is currently not working with sass imports.

**Comment by vlad0** [comment]
Should we execute build command and then run `ng-xi18n` against the build? 

I went at angular-cli(line 204) but it doesn't seem we can setup a path to other folders.
https://github.com/angular/angular-cli/blob/88131a08fd39eab5fc49dfce952207ee826bc8ef/packages/webpack/src/plugin.ts

**Comment by ruffiem** [comment]
@vlad0 Eventually, running ng-xi18n command against the build would have no effect since it serves compiled files.

Anyone tried to exclude CSS files using filesGlob ?

**Comment by sebastianhaas** [comment]
I'm actually doing something like 
```shell
# 1) Comment out all styleUrl declarations
for i in src/app/*.component.ts; do
    sed -i '' 's/\(styleUrls:.*\)/\/\/ \1/g' "$i";
done

# 2) Run ng-x18n
npm run i18n

# 3) undo 1) [...]
```
so I can confirm that ng-x18n works fine without the style declarations. I was able to get rid of most sass errors by changing sass imports from `~@angular/foo/theme` to, for instance, `../../node_modules/@angular/foo/_theme.scss`, but even then, there is https://github.com/angular/material2/issues/1348 blocking sass compilation.

**Comment by lucrod1** [comment]
@sebastianhaas You did it undo comment `// styleUrls:` for `styleUrls:` ?
Could you send us the code?

**Comment by tobiaseisenschenk** [comment]
I am not sure if it is related since I dont use i18n, but I have the same issue otherwise. I enabled Sass support in my angular seed fork by following the respective wiki entry. Hence my project-wide sass files are within this path: src>client>assets>sass>main.scss and are compiled correctly. Interestingly only some (not all) of my **component-specific-scss** files cause an error. Chrome tells me 'can not find xxx.component.css' even though the compiled css file exists under dist>temp. (Yes I did check for spelling mistakes ;-))
Is this possibly related to the above described issue, am I obviously doing something wrong or should I just head over to SO? Thanks!
